### PR TITLE
Use "KeyboardEvent.key" instead of "Keyboard.keyCode"

### PR DESCRIPTION
**Pull request by skayo**
**What kind of change does this PR introduce?**
- [ ] Bug Fix
- [ ] Feature
- [X] Refactoring
- [ ] Style
- [ ] Build
- [ ] Chore
- [ ] Documentation
- [ ] CI

**Did you add tests for your changes?**
- [X] Yes, my code is well tested
- [ ] Not relevant

**If relevant, did you update the documentation?**
- [ ] Yes, I've updated the documentation
- [X] Not relevant

**Summary**
`KeyboardEvent.keyCode` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) and should not be used anymore.  
Browsers could drop support anytime and it's not recommended to use it.

**Does this PR introduce a breaking change?**
I don't think so.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -742,7 +742,11 @@ export function useDropzone(options = {}) {
         return;
       }
 
-      if (event.keyCode === 32 || event.keyCode === 13) {
+      if (
+        (event.key === 'Enter' || event.key === ' ') ||
+        (event.code === 'Enter' || event.code === 'Space') ||
```
Comment (rolandjitsu) [inline src/index.js:747]: From MDN:
```
For example, the code returned is "KeyQ" for the Q key on a QWERTY layout keyboard, but the same code value also represents the ' key on Dvorak keyboards and the A key on AZERTY keyboards. That makes it impossible to use the value of code to determine what the name of the key is to users if they're not using an anticipated keyboard layout.
```

So `evt.code` is also not reliable. But `evt.key` seems like it is. So I don't think we need this one.

Comment (skayo) [reply src/index.js:747]: Alright, just added it cause I wasn't sure

File: src/index.js [hunk 2]
```diff
@@ -742,7 +742,10 @@ export function useDropzone(options = {}) {
         return;
       }
 
-      if (event.keyCode === 32 || event.keyCode === 13) {
+      if (
```
Comment (rolandjitsu) [inline src/index.js:745]: We'll obviously need some unit tests. There's already a bunch of them testing the `keyCode`.

Comment (skayo) [reply src/index.js:745]: I'll look into it when I have the time!

**Comment by skayo** [comment]
@rolandjitsu I'm a bit unsure about how exactly we should test this properly. I noticed that most people are using `key` and `keyCode` simultaneously in `fireEvent.keyDown`. For example:
```
fireEvent.keyDown(dropzone, {
  key: 'Enter',
  keyCode: 13,
});
```

Should I also do this? Or write separate test for each `keyCode` and `key`?

**Comment by skayo** [comment]
@rolandjitsu added tests now. Could you take a look?

**Comment by rolandjitsu** [comment]
> @rolandjitsu added tests now. Could you take a look?

Thanks. I'll have a look this weekend.

**Review by rolandjitsu — APPROVED** [review]

**Pull request merged**